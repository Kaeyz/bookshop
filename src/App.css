body {
  height: 100vh;
}

.book_list{
  margin-bottom: 5vh;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fill, minmax(12rem, 1fr));
}

.img{
  height: 23em;
}

.cart_section {
  margin-top: 5vh;
  margin-bottom: 1vh;
  width: 100%;
  display: grid;
  justify-content: center;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 10px;
}
