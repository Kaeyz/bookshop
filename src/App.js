import React from 'react';
import './App.css';

import BookList from "./components/BooksList";


function App() {
  return (
    <div>
      <nav className="navbar navbar-dark bg-dark text-light">
       <h5>Harry's Adventure Series</h5>
      </nav>

      <BookList />

      <footer className="card-footer text-light bg-dark">
        <p className="text-center">2019</p>
      </footer>
    </div>
  );
}

export default App;
