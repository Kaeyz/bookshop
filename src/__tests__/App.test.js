import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";


import App from "../App";
import BookList from "../components/BooksList";

describe('app renders without crashing', () => {
  const app = shallow(<App />);

  it("renders the header correctly", () => {
    const header = <h5>Harry's Adventure Series</h5>;
    expect(app.contains(header)).toEqual(true);
  });

  it("renders the footer correctly", () => {
    const footer = <footer className="card-footer text-light bg-dark">
      <p className="text-center">2019</p>
    </footer>;
    expect(app.contains(footer)).toEqual(true);
  });

  it("renders the bookList component", () => {
    const bookList = <BookList />;
    expect(app.contains(bookList)).toEqual(true);
  });
});


describe('testing app with snapshot', () => {
  const app = shallow(<App />);
  it('must match snapshot', () => {
    expect(toJSON(app)).toMatchSnapshot()
  })
})