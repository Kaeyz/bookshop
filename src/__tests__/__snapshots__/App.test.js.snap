// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing app with snapshot must match snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-dark bg-dark text-light"
  >
    <h5>
      Harry's Adventure Series
    </h5>
  </nav>
  <BooksList />
  <footer
    className="card-footer text-light bg-dark"
  >
    <p
      className="text-center"
    >
      2019
    </p>
  </footer>
</div>
`;
